src/main/java/frc/robot/subsystems/SuperStructure.kt

```diff
@@ -176,4 +181,79 @@ object SuperStructure {
+                            FieldGeometry.CORAL_STATIONS,
+                            Chassis.state.Pose.translation,
+                        )
+                        .meters + PIVOT_TO_CENTER - CLAW_HORIZONTAL
+
+                // Trig calculation on target height and distance away, then account for Pivot bias
+                // (zeroed below horizontal)
+                val targetPivotAngle: Angle =
+                    atan(INTAKE_HEIGHT.inches / baseDist.inches).radians + PIVOT_BIAS
+
+                // Pythagorean theorem, then subtract retracted height of elevator to give the
+                // necessary extension
+                val targetElevatorHeight: Distance =
+                    sqrt(baseDist.inches.pow(2) + INTAKE_HEIGHT.inches.pow(2)).inches -
+                        ELEVATOR_LENGTH
+
+                // Needs to decrease by the amount the Pivot angle decreases relative to the base
+                // preset
+                val targetWristAngle: Angle =
+                    RobotState.CoralStation.wristAngle -
+                        (RobotState.CoralStation.pivotAngle - targetPivotAngle)
+
+                // Distance from frame to coral station; should not extend if > 18in
+                val frameToStation: Distance =
+                    FieldGeometry.distanceToClosestLine(
+                            FieldGeometry.CORAL_STATIONS,
+                            Chassis.state.Pose.translation,
+                        )
+                        .meters - 14.inches
+
+                //            SmartDashboard.putNumber("Raw Station Dist", )
+                //            SmartDashboard.putNumber("height value in", INTAKE_HEIGHT.inches)
+                //            SmartDashboard.putNumber("base dist m", baseDist.meters)
+                //            SmartDashboard.putNumber("tgt pivot deg", targetPivotAngle.degrees)
+                //            SmartDashboard.putNumber("tgt elevator in",
+                // targetElevatorHeight.inches)
+                //            SmartDashboard.putNumber("tgt wrist deg", targetWristAngle.degrees)
+
+                // Waiting for Pivot not necessary at shorter (legal) distances
+                //            Pivot.goToRaw(targetPivotAngle)
+                //                .andWait { Pivot.atPosition }
+                //                .andThen(
+                //                    Elevator.goToRaw(targetElevatorHeight)
+                //                        .alongWith(Wrist.goToRaw(targetWristAngle)))
+
+                // Extend if it doesn't violate extension rule
+                ConditionalCommand(
+                    Commands.parallel(
+                        Pivot.goToRaw(targetPivotAngle),
+                        Elevator.goToRaw(targetElevatorHeight),
+                        Wrist.goToRaw(targetWristAngle),
+                    ),
+                    Commands.idle(),
+                ) {
+                    frameToStation <= 18.inches
```

Comment:
I think you can remove the deferred part, and just throw all the logic to calculate frameToStation in here, right?